src/rendering/renderers/gpu/texture/GpuTextureSystem.ts

```diff
@@ -70,6 +70,10 @@ export class GpuTextureSystem implements System, CanvasGenerator
 
     public initSource(source: TextureSource): GPUTexture
     {
+        let gpuTexture = this._gpuSources[source.uid];
```

Comment:
heyhey! whilst this is a fixe, the deeper issue is that initSource should not be called if the texture already exists.